### ISSUE TITLE
anyway to display disconnected commits in a branch

### ISSUE THREAD

**Issue by clime** [ostreedev/ostree#2065]
Hello,

currently in our workflow, on host machines, we get commits that do not have any child preset (because those children were only intermediate non-production builds). That means when somebody does `ostree log master` (`master` is the only branch on target machine containing all the commits), always only a single commit is shown and all the other are not shown in the log output because they are not direct descendants of the first commits. Is there a way to get more "full" view, that would really display all the commits in the branch even if they are not direct descendants of each other?

Thank you

**Comment by clime** [comment]
By the way, you did really a good job on OSTree. Thank you for that.

**Comment by cgwalters** [comment]
> currently in our workflow, on target machines, we get commits that do not have any child present (because those children were only intermediate non-production builds).

Yes...this topic has been a huge tricky problem to solve for us since we introduced Fedora Atomic Host (and now Fedora/RHEL CoreOS).  One of the most thorny problems here is that for people using ostree to build operating systems (as opposed to e.g. flatpak) you need to build/ship/manage/version things like ISO images and AMIs.  And those need history/versioning too - and one gets into questions around whether you want to allow them to drift.

For [coreos-assembler](https://github.com/coreos/coreos-assembler/) we ended up with a simple "json in S3" as a versioning model.

For RHEL CoreOS we simply don't use the ostree history because the expectation is that OS updates are cluster driven.

For FCOS...see https://github.com/coreos/fedora-coreos-tracker/issues/199 - I think we ended up importing all commits?

**Comment by dbnicholson** [comment]
Stumbled onto this old issue.

We had the same thing at Endless and I ended up writing something similar to `flatpak build-commit-from` that takes the new build commit and essentially rebases it on the current head commit. This is part of the receive side of ostree-push in https://github.com/dbnicholson/ostree-push/blob/master/otpush/receive.py#L146.

We use the same procedure during release to rebase the "to be released" build commit onto the last released commit. That way our production history is intact while the intermediate builds are on some other branch (actually a separate repo, but the concept is the same).
